pinot-core/src/main/java/org/apache/pinot/core/startree/operator/StarTreeFilterOperator.java

```diff
@@ -267,19 +206,19 @@ private StarTreeResult traverseStarTree() {
     // Use BFS to traverse the star tree
     Queue<SearchEntry> queue = new LinkedList<>();
     queue.add(new SearchEntry(starTreeRootNode, _predicateEvaluatorsMap.keySet(), _groupByColumns));
-    while (!queue.isEmpty()) {
-      SearchEntry searchEntry = queue.remove();
+    SearchEntry searchEntry;
+    while ((searchEntry = queue.poll()) != null) {
```

Comment:
while (searchEntry = queue.poll() != null) {

Reply:
The parentheses can not be removed, or it will try to assign the boolean value to `searchEntry`

Reply:
got it